referenceframe/urdf.go

```diff
@@ -0,0 +1,313 @@
+			}
+
+			// Slightly different limits handling for continuous, revolute, and prismatic joints
+			if jointElem.Type == "continuous" {
+				thisJoint.Type = "revolute" // Currently, we treate a continuous joint as a special case of the revolute joint
+				thisJoint.Min, thisJoint.Max = math.Inf(-1), math.Inf(1)
+			} else if jointElem.Type == "prismatic" {
+				thisJoint.Min, thisJoint.Max = metersToMM(jointElem.Limit.Lower), metersToMM(jointElem.Limit.Upper)
+			} else if jointElem.Type == "revolute" {
+				thisJoint.Min, thisJoint.Max = utils.RadToDeg(jointElem.Limit.Lower), utils.RadToDeg(jointElem.Limit.Upper)
+			} else {
+				return nil, err
+			}
+
+			mc.Joints = append(mc.Joints, thisJoint)
+
+			// Generate child link translation and orientation data, which is held by this joint per the URDF design
+			childXYZ := convStringAttrToFloats(jointElem.Origin.XYZ)
+			childRPY := convStringAttrToFloats(jointElem.Origin.RPY)
+			childEA := spatial.EulerAngles{Roll: childRPY[0], Pitch: childRPY[1], Yaw: childRPY[2]}
+			childOrient, err := spatial.NewOrientationConfig(childEA.EulerAngles())
+
+			// Note the conversion from meters to mm
+			childLink.Translation = spatial.TranslationConfig{
+				metersToMM(childXYZ[0]),
+				metersToMM(childXYZ[1]),
+				metersToMM(childXYZ[2]),
+			}
+			childLink.Orientation = *childOrient
+
+			if err != nil {
+				return nil, err
+			}
+		case "fixed":
+			// Handle fixed joint -> static link conversion instead of adding to Joints[]
+			thisLink := JsonLink{ID: jointElem.Name, Parent: jointElem.Parent.Link}
+
+			linkXYZ := convStringAttrToFloats(jointElem.Origin.XYZ)
+			linkRPY := convStringAttrToFloats(jointElem.Origin.RPY)
+			linkEA := spatial.EulerAngles{Roll: linkRPY[0], Pitch: linkRPY[1], Yaw: linkRPY[2]}
+			linkOrient, err := spatial.NewOrientationConfig(linkEA.AxisAngles())
+
+			// Note the conversion from meters to mm
+			thisLink.Translation = spatial.TranslationConfig{
+				metersToMM(linkXYZ[0]),
+				metersToMM(linkXYZ[1]),
+				metersToMM(linkXYZ[2]),
+			}
+			thisLink.Orientation = *linkOrient
+
+			if err != nil {
+				return nil, err
+			}
+
+			mc.Links = append(mc.Links, thisLink)
+		default:
+			return nil, NewUnsupportedJointTypeError(jointElem.Type)
+		}
+
+		if err != nil {
+			return nil, err
+		}
+
+		mc.Links = append(mc.Links, childLink)
+	}
+
+	// Handle links
+	for _, linkElem := range urdf.Links {
+		// Skip any world links
+		if linkElem.Name == "world" {
+			continue
+		}
+
+		// Find matching links which already exist, take care of geometry if collision elements are detected
+		hasCollision := len(linkElem.Collision) > 0
+		for idx, prefabLink := range mc.Links {
+			if prefabLink.ID == linkElem.Name && hasCollision {
+				geoCfg, _ := createConfigFromCollision(linkElem)
+				mc.Links[idx].Geometry = geoCfg
+				break
+			}
+		}
+
+		// In the event the link does not already exist in the ModelConfig, we will have to generate it now
```

Comment:
when would this already exist in the ModelConfig?

Reply:
On L131, we create a child link prior to parsing through the URDF Links themselves.